src/tests/src/solver/infeasible-problem-analysis/test-unfeasible-problem-analyzer.cpp

```diff
@@ -0,0 +1,207 @@
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+#define WIN32_LEAN_AND_MEAN
+#define BOOST_TEST_MODULE unfeasible_problem_analyzer
+#define BOOST_TEST_DYN_LINK
+
+#include <boost/test/unit_test.hpp>
+#include <boost/test/data/test_case.hpp>
+#include <boost/test/data/dataset.hpp>
+
+#include "infeasible-problem-analysis/unfeasible-pb-analyzer.h"
+#include "infeasible-problem-analysis/variables-bounds-consistency.h"
+#include "infeasible-problem-analysis/constraint-slack-analysis.h"
+
+namespace bdata = boost::unit_test::data;
+
+using operations_research::MPSolver;
+
+using Antares::Optimization::VariablesBoundsConsistency;
+using Antares::Optimization::VariableBounds;
+using Antares::Optimization::ConstraintSlackAnalysis;
+using Antares::Optimization::UnfeasiblePbAnalyzer;
+using Antares::Optimization::UnfeasibilityAnalysis;
+
+
+bool variableEquals(const VariableBounds& lhs, const VariableBounds& rhs)
+{
+    return lhs.name == rhs.name &&
+        lhs.lowBound == rhs.lowBound &&
+        lhs.upBound == rhs.upBound;
+}
+
+/*!
+ * Analysis mock, used to assess which step has been run by the analyzer
+ */
+class AnalysisMock : public UnfeasibilityAnalysis
+{
+public:
+    AnalysisMock(bool shouldDetectCause, bool& hasRun, bool& hasPrinted):
+            shouldDetectCause_(shouldDetectCause),
+            hasRun_(hasRun),
+            hasPrinted_(hasPrinted)
+    {}
+
+    ~AnalysisMock() override = default;
+
+    void run(operations_research::MPSolver *problem) override
+    {
+        hasRun_ = true;
+        hasDetectedInfeasibilityCause_ = shouldDetectCause_;
+    }
+
+    void printReport() const override
+    {
+        hasPrinted_ = true;
+    }
+
+    std::string title() const override
+    {
+        return "mock";
+    }
+
+private:
+    bool shouldDetectCause_;
+    bool& hasRun_;
+    bool& hasPrinted_;
+};
+
+
+BOOST_AUTO_TEST_SUITE(unfeasible_problem_analyzer)
+
+BOOST_AUTO_TEST_CASE(test_problem_analyzer)
+{
+    bool hasRun1 = false;
+    bool hasPrinted1 = false;
+    bool hasRun2 = false;
+    bool hasPrinted2 = false;
+    std::vector<std::unique_ptr<UnfeasibilityAnalysis>> analysis;
+    analysis.push_back(std::make_unique<AnalysisMock>(false, hasRun1, hasPrinted1));
+    analysis.push_back(std::make_unique<AnalysisMock>(true, hasRun2, hasPrinted2));
+    std::unique_ptr<MPSolver> problem(MPSolver::CreateSolver("GLOP"));
+
+    UnfeasiblePbAnalyzer analyzer(std::move(analysis));
```

Comment:
Yep, necessary for the compiler, because the vector is move-only (because its content is move-only).
without `std::move` I guess the compiler tries to perform a copy.